src/main/java/teammates/ui/webapi/Action.java

```diff
@@ -89,7 +89,13 @@ public void setLogsProcessor(LogsProcessor logsProcessor) {
     /**
      * Checks if the requesting user has sufficient authority to access the resource.
      */
-    void checkAccessControl() {
+    void checkAccessControl() throws UnauthorizedAccessException {
+        String userParam = getRequestParamValue(Const.ParamsNames.USER_ID);
+        if (userInfo != null && userParam != null && !userInfo.isAdmin && !userInfo.id.equals(userParam)) {
+            throw new UnauthorizedAccessException("User " + userInfo.id
+                    + " is trying to masquerade as " + userParam + " without admin permission.");
```

Comment:
Will it be better if we throw an `UnauthorizedAccessException` as long as either `!userInfo.isAdmin` _or_ `!userInfo.id.equals(userParam)`?

Reply:
Both conditions imply `userInfo != null`. Non-existent `userParam` means no intention to masquerade, malicious or not.